In [2]:
import pandas as pd
import numpy as np

In [6]:
data= pd.read_csv('marketing_data.csv')
data

,ID,Year_Birth,Education,Marital_Status,Income,Kidhome,Teenhome,Dt_Customer,Recency,MntWines,...,NumStorePurchases,NumWebVisitsMonth,AcceptedCmp3,AcceptedCmp4,AcceptedCmp5,AcceptedCmp1,AcceptedCmp2,Response,Complain,Country
0,1826,1970,Graduation,Divorced,84835.0,0,0,2014-06-16,0,189,...,6,1,0,0,0,0,0,1,0,Spain
1,1,1961,Graduation,Single,57091.0,0,0,2014-06-15,0,464,...,7,5,0,0,0,0,1,1,0,Canada
2,10476,1958,Graduation,Married,67267.0,0,1,2014-05-13,0,134,...,5,2,0,0,0,0,0,0,0,USA
3,1386,1967,Graduation,Together,32474.0,1,1,2014-05-11,0,10,...,2,7,0,0,0,0,0,0,0,Australia
4,5371,1989,Graduation,Single,21474.0,1,0,2014-04-08,0,6,...,2,7,1,0,0,0,0,1,0,Spain
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
2235,10142,1976,PhD,Divorced,66476.0,0,1,2013-03-07,99,372,...,11,4,0,0,0,0,0,0,0,USA
2236,5263,1977,2n Cycle,Married,31056.0,1,0,2013-01-22,99,5,...,3,8,0,0,0,0,0,0,0,Spain
2237,22,1976,Graduation,Divorced,46310.0,1,0,2012-12-03,99,185,...,5,8,0,0,0,0,0,0,0,Spain
2238,528,1978,Graduation,Married,65819.0,0,0,2012-11-29,99,267,...,10,3,0,0,0,0,0,0,0,India


In [11]:
data.columns
data.info()
data.head()
data.tail()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2240 entries, 0 to 2239
Data columns (total 28 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   ID                   2240 non-null   int64  
 1   Year_Birth           2240 non-null   int64  
 2   Education            2240 non-null   object 
 3   Marital_Status       2240 non-null   object 
 4    Income              2216 non-null   float64
 5   Kidhome              2240 non-null   int64  
 6   Teenhome             2240 non-null   int64  
 7   Dt_Customer          2240 non-null   object 
 8   Recency              2240 non-null   int64  
 9   MntWines             2240 non-null   int64  
 10  MntFruits            2240 non-null   int64  
 11  MntMeatProducts      2240 non-null   int64  
 12  MntFishProducts      2240 non-null   int64  
 13  MntSweetProducts     2240 non-null   int64  
 14  MntGoldProds         2240 non-null   int64  
 15  NumDealsPurchases    2240 non-null   i

,ID,Year_Birth,Education,Marital_Status,Income,Kidhome,Teenhome,Dt_Customer,Recency,MntWines,...,NumStorePurchases,NumWebVisitsMonth,AcceptedCmp3,AcceptedCmp4,AcceptedCmp5,AcceptedCmp1,AcceptedCmp2,Response,Complain,Country
2235,10142,1976,PhD,Divorced,66476.0,0,1,2013-03-07,99,372,...,11,4,0,0,0,0,0,0,0,USA
2236,5263,1977,2n Cycle,Married,31056.0,1,0,2013-01-22,99,5,...,3,8,0,0,0,0,0,0,0,Spain
2237,22,1976,Graduation,Divorced,46310.0,1,0,2012-12-03,99,185,...,5,8,0,0,0,0,0,0,0,Spain
2238,528,1978,Graduation,Married,65819.0,0,0,2012-11-29,99,267,...,10,3,0,0,0,0,0,0,0,India
2239,4070,1969,PhD,Married,94871.0,0,2,2012-09-01,99,169,...,4,7,0,1,1,0,0,1,0,Canada


In [13]:
data.columns = data.columns.str.strip()
print(data.isnull().sum())

ID                      0
Year_Birth              0
Education               0
Marital_Status          0
Income                 24
Kidhome                 0
Teenhome                0
Dt_Customer             0
Recency                 0
MntWines                0
MntFruits               0
MntMeatProducts         0
MntFishProducts         0
MntSweetProducts        0
MntGoldProds            0
NumDealsPurchases       0
NumWebPurchases         0
NumCatalogPurchases     0
NumStorePurchases       0
NumWebVisitsMonth       0
AcceptedCmp3            0
AcceptedCmp4            0
AcceptedCmp5            0
AcceptedCmp1            0
AcceptedCmp2            0
Response                0
Complain                0
Country                 0
dtype: int64


# income column has missing values so we will impute with median income

In [22]:
mean_income= data['Income'].mean()
mean_income
data['Income'].fillna(mean_income, inplace=True)
print(data.isnull().sum())

ID                     0
Year_Birth             0
Education              0
Marital_Status         0
Income                 0
Kidhome                0
Teenhome               0
Dt_Customer            0
Recency                0
MntWines               0
MntFruits              0
MntMeatProducts        0
MntFishProducts        0
MntSweetProducts       0
MntGoldProds           0
NumDealsPurchases      0
NumWebPurchases        0
NumCatalogPurchases    0
NumStorePurchases      0
NumWebVisitsMonth      0
AcceptedCmp3           0
AcceptedCmp4           0
AcceptedCmp5           0
AcceptedCmp1           0
AcceptedCmp2           0
Response               0
Complain               0
Country                0
dtype: int64


C:\Users\sinha\AppData\Local\Temp\ipykernel_28512\3241422778.py:3: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  data['Income'].fillna(mean_income, inplace=True)


# Calculated KPI's

In [42]:
# Totalspend
data['Total_spend'] = (data['MntWines'] + data['MntFruits'] 
                       + data['MntMeatProducts'] + data['MntFishProducts'] + 
                       data['MntSweetProducts'] + data['MntGoldProds'])
data['Total_spend']

# Totalpurchase
data['Total_purchase'] = (data['NumWebPurchases']+ data['NumCatalogPurchases'] + 
                          data['NumStorePurchases'])

# SpendperPurchase
data['Spend/Purchase'] = (data['Total_spend'] / data['Total_purchase'].replace(0, 1)).round(2)

# Campaigns accepted
data['CampaignsAccepted'] = (data['AcceptedCmp1'] + data['AcceptedCmp2'] + 
                              data['AcceptedCmp3'] + data['AcceptedCmp4'] + 
                              data['AcceptedCmp5'] + data['Response'])

#verify new columns
print(data[['Total_spend','Total_purchase',
            'Spend/Purchase','CampaignsAccepted']].describe().round(2))

       Total_spend  Total_purchase  Spend/Purchase  CampaignsAccepted
count      2240.00         2240.00         2240.00            2240.00
mean        605.80           12.54           38.12               0.45
std         602.25            7.21           45.90               0.89
min           5.00            0.00            2.67               0.00
25%          68.75            6.00           13.00               0.00
50%         396.00           12.00           29.64               0.00
75%        1045.50           18.00           49.13               1.00
max        2525.00           32.00         1679.00               5.00


In [44]:
# Ghost customers. Customers who spent 0
print(data[data['Total_purchase'] == 0].shape[0], "customers with 0 purchases")


# Top 5 spenders
print(data.nlargest(5, 'Total_spend')[['Income','Total_spend','CampaignsAccepted']])

6 customers with 0 purchases
       Income  Total_spend  CampaignsAccepted
671   90638.0         2525                  1
672   90638.0         2525                  2
1404  87679.0         2524                  4
1026  75759.0         2486                  4
1864  69098.0         2440                  0


## Insights
- There are  only 6 out of 2,240 customers never purchased. That's just 0.27% — your customer base is highly active. You can safely ignore or drop these 6 rows in Phase 2.
- Rows 671 and 672 have same income, need to dig deeper if this is a duplicate entry.
- Row 1404 is ideal customer — high income, highest campaign response (4/6), high spend. This is the profile you want to target more of.
- Row 1864 is a biggest puzzle — spends $2,440 but accepted ZERO campaigns. This customer shops without any marketing push. Understanding why is a key Phase 2 SQL question.